src/Microsoft.TemplateEngine.Orchestrator.RunnableProjects/RunnableProjectConfig.cs

```diff
@@ -646,30 +646,33 @@ private static IReadOnlyList<string> TryReadConfigFromFile(IReadOnlyList<string>
 
         private IReadOnlyList<IReplacementTokens> ProduceSymbolFilenameReplacements()
         {
-            List<IReplacementTokens> filenameReplacements = new List<IReplacementTokens>();
+            List<IReplacementTokens> filenameReplacements = new();
             if (!_configuration.Symbols.Any())
             {
                 return filenameReplacements;
             }
 
-            foreach (BaseReplaceSymbol symbol in _configuration.Symbols.OfType<BaseReplaceSymbol>().Where(s => !string.IsNullOrWhiteSpace(s.FileRename)))
+            foreach (BaseReplaceSymbol symbol in _configuration.Symbols.OfType<BaseReplaceSymbol>())
             {
                 if (symbol is BaseValueSymbol p)
                 {
-                    foreach (string formName in p.Forms.GlobalForms)
+                    if (!string.IsNullOrWhiteSpace(symbol.FileRename))
                     {
-                        if (_configuration.Forms.TryGetValue(formName, out IValueForm valueForm) && p.FileRename != null)
+                        foreach (string formName in p.Forms.GlobalForms)
                         {
-                            string symbolName = symbol.Name + "{-VALUE-FORMS-}" + formName;
-                            string? processedFileReplacement = valueForm.Process(p.FileRename, _configuration.Forms);
-                            if (!string.IsNullOrEmpty(processedFileReplacement))
+                            if (_configuration.Forms.TryGetValue(formName, out IValueForm valueForm))
                             {
-                                GenerateFileReplacementsForSymbol(processedFileReplacement!, symbolName, filenameReplacements);
+                                string symbolName = symbol.Name + "{-VALUE-FORMS-}" + formName;
+                                string processedFileReplacement = valueForm.Process(p.FileRename!, _configuration.Forms);
```

Comment:
nit: I'd use `symbol` instead of `p` here for consistency and clarity. Reading this code I was first confused why we made assumption about `p.FileRename` not being null